src/globals.js

```diff
@@ -40,7 +40,7 @@ function joiner(sep) {
 function globals() {
     return {
         range: function(start, stop, step) {
-            if(!stop) {
+            if(!stop && stop !== 0) {
```

Comment:
Yep, I appreciate the motivation to preserve maximum back-compat. But back-compat for undocumented edge-cases always has to be weighed against the likelihood of someone actually relying on the current behavior (for reasonable reasons), and I just don't see it here.